### PR TITLE
include role in addUser(ToCurrentOrg)

### PR DESCRIPTION
**Pull request by jeremyzitomer-usds**
## Related Issue or Background Info

- Added user role as parameter in addUser() and addUserToCurrentOrg(), backward-compatibly; updated tests accordingly

### CODE REVIEW

**Review by timothybest-usds — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/apiuser/ApiUserMutationResolver.java [hunk 1]
```diff
@@ -31,9 +28,14 @@ public User addUser(
...
             String suffix,
             String email,
-            String organizationExternalID
+            String organizationExternalID,
+            OrganizationRole role
                 ) {
-        UserInfo user = _us.createUser(email, firstName, middleName, lastName, suffix, organizationExternalID);
+        // For backward compatibility
+        if (role == null) {
+            role = OrganizationRole.USER;
```
Comment (nickscialli-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/apiuser/ApiUserMutationResolver.java:36]: 👍

**Review by nickscialli-usds — COMMENTED** [review]
Seems good to me but don't feel comfortable "approving" BE changes

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java [hunk 2]
```diff
@@ -88,21 +89,28 @@ public LiveOktaRepository(AuthorizationProperties authorizationProperties,
...
+        
+        for (OrganizationRole r : roles) {
+            String groupName = generateGroupName(organizationExternalId, r);
 
-        // Okta SDK's way of getting a group by group name
-        GroupList groups = _client.listGroups(groupName, null, null);
-        if (!groups.iterator().hasNext()) {
-            throw new IllegalGraphqlArgumentException("Cannot add Okta user to nonexistent group");
+            // Okta SDK's way of getting a group by group name
+            GroupList groups = _client.listGroups(groupName, null, null);
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:99]: It seems like Okta _ought_ to support getting all our relevant groups in one go and then iterating over them, which might improve performance as well. Not a blocker here, but we should probably have a collection of Okta performance issues (like #927 or possibly just expanding that issue) to address as we have time.

**Review by benwarfield-usds — APPROVED** [review]
:shipit:

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java (hunk 2 above)
Comment (jeremyzitomer-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:99]: expanding https://github.com/CDCgov/prime-simplereport/issues/927 accordingly

**Pull request merged**